packages/overture-schema-core/src/overture/schema/core/json_schema.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any, get_origin
+
+from pydantic import BaseModel
+
+from ._cache import get_type_adapter
+
+
+def json_schema(models: type[BaseModel] | Any) -> dict[str, Any]:
+    """Generate JSON schema for a Pydantic model or union of models.
+
+    Args:
+        models: Either a Pydantic BaseModel class or a union type (possibly
+                annotated with discriminator information) of BaseModels.
+
+    Returns:
+        dict: JSON schema representation of the model(s).
+
+    Raises:
+        TypeError: If models is not a BaseModel or union type.
+    """
+    if isinstance(models, type) and issubclass(models, BaseModel):
+        return models.model_json_schema()
+
+    if get_origin(models) is not None:
+        adapter = get_type_adapter(models)
+        return adapter.json_schema()
```

Comment:
Use case?

Reply:
Unions: `RoadSegment | RailSegment | WaterSegment` (the cleanest way I could find to model conditional groups of properties) and to produce a unified schema for backward-compatibility. The generated JSON Schema splits these with a `oneOf`: https://github.com/OvertureMaps/schema/blob/f6d4a6bbf50aa673d80dcfd9474692e9e112d8f5/packages/overture-schema-transportation-theme/tests/segment_baseline_schema.json#L1735-L1745